lib/pure/asyncfile.nim

```diff
@@ -300,14 +300,18 @@ proc readLine*(f: AsyncFile): Future[string] {.async.} =
   ## Reads a single line from the specified file asynchronously.
   result = ""
   while true:
-    var c = await read(f, 1)
+    let c = await read(f, 1)
+    if c.len == 0:
+      # EOF
+      raise newException(EOFError, "EOF reached")
     if c[0] == '\c':
-      c = await read(f, 1)
+      # Found a CR, eat the NL
```

Comment:
\L isn't a given here. You can't just discard it and assume that is the case.

Reply:
And? The PR is meant to fix the `IndexError`, I don't care enough about async files to fix this shit

Reply:
Fixing one bug and introducing another isn't productive.

Reply:
> Fixing one bug and introducing another isn't productive.

What is your implementation" for the readLine function?

Reply:
> Fixing one bug and introducing another isn't productive.

Oh come the fuck on, you're not even trying.

```nim
  while true:
    var c = await read(f, 1)
    if c[0] == '\c':
      c = await read(f, 1)
      break
    if c[0] == '\L' or c == "":
      break
    else:
      result.add(c)
```

...

```nim
    if c[0] == '\c':
      c = await read(f, 1)
      break
```

...

```nim
break
```

> **introducing another**

Man, you're the one who added this, step up your blame game :point_right: :sunglasses: :point_right:

Reply:
Fair enough. I misread the code, my bad. But there is absolutely no need to start cursing.